notdiamond/llms/providers.py

```diff
@@ -23,6 +23,10 @@ class NDLLMProviders(Enum):
         GPT_4o_MINI_2024_07_18 (NDLLMProvider): refers to 'gpt-4o-mini-2024-07-18' model by OpenAI
         GPT_4o_MINI (NDLLMProvider): refers to 'gpt-4o-mini' model by OpenAI
         GPT_4_0125_PREVIEW (NDLLMProvider): refers to 'gpt-4-0125-preview' model by OpenAI
+        o1_PREVIEW (NDLLMProvider): refers to 'o1-preview' model by OpenAI
+        o1_PREVIEW_2024_09_12 (NDLLMProvider): refers to 'o1-preview-2024-09-12' model by OpenAI
+        o1_MINI (NDLLMProvider): refers to 'o1-mini' model by OpenAI
+        o1_MINI_2024_09_12 (NDLLMProvider): refers to 'o1-mini-2024-09-12' model by OpenAI
```

Comment:
As an Enum I'd prefer to see these capitalized by convention:

```suggestion
        O1_PREVIEW (NDLLMProvider): refers to 'o1-preview' model by OpenAI
        O1_PREVIEW_2024_09_12 (NDLLMProvider): refers to 'o1-preview-2024-09-12' model by OpenAI
        O1_MINI (NDLLMProvider): refers to 'o1-mini' model by OpenAI
        O1_MINI_2024_09_12 (NDLLMProvider): refers to 'o1-mini-2024-09-12' model by OpenAI
```

Reply:
Done